### PR TITLE
docs: Add `params` option to reference docs so that non-js client libs can add parameters

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

Something I should have done a looong time ago. 

Until now, there is no great way to add params on non-js client libs' reference docs. This PR introduces a new `params` parameter on the spec yaml file so that non-js client libs can have parameter definitions in the reference docs, and makes it aligned with the js reference docs. 

Screenshot 2024-03-02 at 18 58 16

## What is the current behavior?

There is the [overrideParams](https://github.com/supabase/supabase/blob/master/apps/docs/components/reference/RefFunctionSection.tsx#L61) parameter that we can add to the spec yaml files, but it only allows us to override what exists on the js client libs, and does not allow us to add any parameters specific to the client libraries. 

## What is the new behavior?

A new `params` parameter is introduced on the spec yaml files, where we can add the parameter definitions. I have done a quick example with the `initializing` function for the Flutter docs. 

## Additional Context

If this PR is merged, I will work on adding other parameters for all the methods and ask the maintainers of other libraries to add parameters.

### CODE REVIEW

**Comment by dshukertjr** [comment]
@MildTomato @saltcod Would you take a look at this PR whenever you have time?

**Comment by kevcodez** [comment]
Think @charislam was also working on a new parser for these?

File: apps/docs/spec/supabase_dart_v2.yml [hunk 1]
```diff
@@ -2,17 +2,14 @@ openref: 0.1
 
 info:
   id: reference/dart
-  title: Supabase Dart Client
+  title: Supabase Flutter Client
```
Comment (dshukertjr) [inline apps/docs/spec/supabase_dart_v2.yml:5]:
```suggestion
  title: Supabase Dart Client
```

File: apps/docs/spec/supabase_dart_v2.yml [hunk 2]
```diff
@@ -2,17 +2,14 @@ openref: 0.1
...
-    Supabase Dart.
-
-  definition: spec/enrichments/tsdoc_v2/combined.json
+    Supabase Flutter.
   slugPrefix: '/'
   specUrl: https://github.com/supabase/supabase/edit/master/apps/docs/spec/supabase_dart_v2.yml
   libraries:
-    - name: 'Dart'
-      id: 'dart'
+    - id: 'dart'
```
Comment (dshukertjr) [inline apps/docs/spec/supabase_dart_v2.yml:12]:
```suggestion
    - name: 'Dart'
    - id: 'dart'
```

File: apps/docs/spec/supabase_dart_v2.yml [hunk 3]
```diff
@@ -2,17 +2,14 @@ openref: 0.1
...
 info:
   id: reference/dart
-  title: Supabase Dart Client
+  title: Supabase Flutter Client
   description: |
 
-    Supabase Dart.
-
-  definition: spec/enrichments/tsdoc_v2/combined.json
+    Supabase Flutter.
```
Comment (dshukertjr) [inline apps/docs/spec/supabase_dart_v2.yml:8]:
```suggestion
    Supabase Dart.
```

File: apps/docs/spec/supabase_dart_v2.yml [hunk 4]
```diff
@@ -6,13 +6,11 @@ info:
...
 
     Supabase Dart.
-
-  definition: spec/enrichments/tsdoc_v2/combined.json
   slugPrefix: '/'
   specUrl: https://github.com/supabase/supabase/edit/master/apps/docs/spec/supabase_dart_v2.yml
   libraries:
     - name: 'Dart'
-      id: 'dart'
+    - id: 'dart'
```
Comment (dshukertjr) [inline apps/docs/spec/supabase_dart_v2.yml:13]:
```suggestion
      id: 'dart'
```

**Review by kangmingtay — APPROVED** [review]

**Review by charislam — APPROVED** [review]
🎉 Love it!

No conflicts with my stuff, I was working on improving the `tsDefinition` extraction, not on specifying types for libraries that don't use it

**Pull request merged**